stdlib/public/runtime/Errors.cpp

```diff
@@ -188,13 +189,37 @@ void swift::dumpStackTraceEntry(unsigned index, void *framePC,
 #endif
 }
 
+#if defined(__ANDROID__)
+struct UnwindContext {
+  void **current;
+  void **end;
+};
+
+static _Unwind_Reason_Code unwindBacktraceCallback(struct _Unwind_Context *context, void *arg) {
+  struct UnwindContext *ctx = static_cast<struct UnwindContext *>(arg);
+  uintptr_t pc = _Unwind_GetIP(context);
```

Comment:
Unfortunately, you need to use the ARM extensions to Unwind to get proper information for `dladdr` to symbolicate.  I'll point you to some sample code that I wrote for this.